Service/GithubApiService.php

```diff
@@ -346,65 +346,59 @@ public function getGithubOwnerInfo(array $item): array
     //     return $publiccode;
     // }
 
-    // /**
-    //  * This function is searching for repositories containing a publiccode.yaml file.
-    //  *
-    //  * @param string $slug
-    //  *
-    //  * @throws GuzzleException
-    //  *
-    //  * @return array|null|Response
-    //  */
-    // public function checkPublicRepository(string $slug)
-    // {
-    //     if ($this->checkGithubKey()) {
-    //         return $this->checkGithubKey();
-    //     }
+    /**
+     * This function is searching for repositories containing a publiccode.yaml file.
+     *
+     * @param string $slug
+     *
+     * @return array|null|Response
+     */
+    public function checkPublicRepository(string $slug)
+    {
+        if (!isset($this->githubApiSource) && !$this->githubApiSource = $this->entityManager->getRepository('App:Gateway')->findOneBy(['location' => 'https://api.github.com'])) {
+            // @TODO Monolog ?
+            isset($this->io) && $this->io->error('Could not find Source: Github API');
+            return [];
+        }
 
-    //     try {
-    //         $response = $this->githubClient->request('GET', 'repos/'.$slug);
-    //     } catch (ClientException $exception) {
-    //         return new Response(
-    //             $exception,
-    //             Response::HTTP_BAD_REQUEST,
-    //             ['content-type' => 'json']
-    //         );
-    //     }
+        $response = $this->callService->call($this->githubApiSource, 'repos/' . $slug);
+        $repository = $this->callService->decodeResponse($this->githubApiSource, $response);
+
+        return $repository['private'];
+    }
 
-    //     $response = json_decode($response->getBody()->getContents(), true);
 
-    //     return $response['private'];
-    // }
 
     /**
-     * Makes sure this action has all the gateway objects it needs.
+     * Makes sure this action has all the gateway objects it needs
      */
     private function getRequiredGatewayObjects()
     {
         // get github source
-        if (!isset($this->githubApiSource) && !$this->githubApiSource = $this->entityManager->getRepository('App:Gateway')->findOneBy(['name' => 'GitHub API'])) {
+        if (!isset($this->githubApiSource) && !$this->githubApiSource = $this->entityManager->getRepository('App:Gateway')->findOneBy(['location' => 'https://api.github.com'])) {
             // @TODO Monolog ?
             isset($this->io) && $this->io->error('Could not find Source: Github API');
-
             return [];
```

Comment:
Waarschijnlijk prima dit, maar in veel gevallen returnen we null, misschien voor consistentie beter